s3transfer/processpool.py

```diff
@@ -108,6 +140,110 @@ def notify_job_complete(self, transfer_id):
             return self._transfer_job_count[transfer_id]
 
 
+class Submitter(multiprocessing.Process):
+    def __init__(self, transfer_config, client_factory,
+                 transfer_monitor, osutil, submitter_queue,
+                 worker_queue):
+        """Submit jobs to fulfill a download file request
+
+        :param transfer_config: Configuration for transfers
+        :param client_factory: ClientFactory for creating S3 clients
+        :param transfer_monitor: Monitor for notifying and retrieving state
+            of transfer
+        :param osutil: OSUtils object to use for os-related behavior when
+            performing the transfer.
+        :param submitter_queue: Queue to retrieve download file jobs
+        :param worker_queue: Queue to submit GetObjectJobs for workers
+            to perform
+        """
+        super(Submitter, self).__init__()
+        self._transfer_config = transfer_config
+        self._client_factory = client_factory
+        self._transfer_monitor = transfer_monitor
+        self._osutil = osutil
+        self._submitter_queue = submitter_queue
+        self._worker_queue = worker_queue
+        self._client = None
+
+    def run(self):
+        # Client are not pickleable so their instantiation cannot happen
+        # in the __init__ for processes that are created under the
+        # spawn method.
+        self._client = self._client_factory.create_client()
+        while True:
+            download_file_job = self._submitter_queue.get()
+            if download_file_job == SHUTDOWN_SIGNAL:
+                return
+            try:
+                self._submit_get_object_jobs(download_file_job)
+            except Exception as e:
+                self._transfer_monitor.notify_exception(
```

Comment:
No it cannot, but it really does not need to know. The exception is passed to the monitor so it can be passed to the end user and have them determine what they need to change in their input parameters to get it to work. In general, they should not have to be aware of the fact that there is a submitter/worker pool.